pkg/stacks/relationship.go

```diff
@@ -48,16 +61,54 @@ type KindlyIdentifier interface {
 	GroupVersionKind() schema.GroupVersionKind
 }
 
+// MultiParentLabelPrefix returns the NS specific prefix of a multi-parent label
+// for resources co-owned by a set of Stacks.
+//
+// This prefix is suitable for identifying resources labeled within a given
+// namespace. The prefix may include a predictable truncation suffix if the
+// namespace exceeds 32 characters. This truncation length permits another 32
+// characters for a (potentially truncated) resource name to be appended to the
+// label.
+func MultiParentLabelPrefix(stackParent metav1.Object) string {
+	ns := stackParent.GetNamespace()
+
+	// guaranteed not to error based on the lengths of values we are supplying
+	truncated, _ := truncate.Truncate(ns, preserveNSLength, truncate.DefaultSuffixLength)
+	return fmt.Sprintf(LabelMultiParentNSFormat, truncated)
+}
+
+// MultiParentLabel returns a label name identifying the namespaced name of the
+// stack resource that co-owns another resource
+//
+// The label returned is based on the MultiParentLabelPrefix, which may include
+// a truncation suffix, and is then potentially truncated again to fit in the
+// complete label length restrictions.
+func MultiParentLabel(stackParent metav1.Object) string {
+	prefix := MultiParentLabelPrefix(stackParent)
+
+	// guaranteed 2 parts based on LabelMultiParentNSFormat
```

Comment:
Are we using that format here? Since we accept input after it's been formatted, and since this is a public method, I'm not sure we can assume that in the long run. But it may be okay to make the assumption today and revisit it later.

Reply:
The `MultiParentLabelPrefix` function relies on the `LabelMultiParentPrefix` constant which includes a `/`.  Namespaces can not contain `/` in their resource names. 

If that ever changes, without the rules of label prefixes changing (they allow a single `/`) we would have problems. We have to agree that the world is flat or we have nothing to stand on.